src/core/qgsofflineediting.cpp

```diff
@@ -1015,7 +1018,8 @@ void QgsOfflineEditing::applyFeaturesRemoved( QgsVectorLayer *remoteLayer, sqlit
   int i = 1;
   for ( QgsFeatureIds::const_iterator it = values.constBegin(); it != values.constEnd(); ++it )
   {
-    QgsFeatureId fid = remoteFid( db, layerId, *it );
+    QgsFeatureId fid = remoteFid( db, layerId, *it, remoteLayer );
+    QgsFeature feature = remoteLayer->getFeature( fid );
     remoteLayer->deleteFeature( fid );
```

Comment:
```suggestion
    remoteLayer->deleteFeature( feature.id() );
```
?
otherwise, the line before is never used?

Reply:
the `QgsFeature feature = remoteLayer->getFeature( fid );` is extra and needs to be removed. Is there a situation where `fid != remoteLayer->getFeature( fid ).id()`, unless `getFeature` returns invalid feature?

Reply:
I hope not! Let's remove the extra line then